sycl/include/sycl/ext/intel/fpga_device_selector.hpp

```diff
@@ -22,6 +22,29 @@ class platform;
 namespace ext {
 namespace intel {
 
+namespace detail {
+// Scores a device by platform name. Uses const char * for the platform name to
```

Comment:
Can't a reference to a string be passed to avoid string copying? 'const std::string &required_platform_name'

Reply:
Yes, but since we define the new selectors as functions rather than classes we would have to create a lambda to capture the strings and I don't see much benefit to that complexity. If you disagree I can change it.

Reply:
I'd prefer to have `std::string_view` over C-style strings, if possible.

Reply:
`std::string_view` seems like a good solution.